collection/collection_map.go

```diff
@@ -205,10 +224,16 @@ func (c *Map) Data() map[string][]string {
 var _ Collection = &Map{}
 
 // NewMap returns a collection of key->[]values
-func NewMap(variable variables.RuleVariable) *Map {
+// If caseSensitiveKeys is false, the keys will be converted to lower case
+// Some keys must be stored using lower case keys, like HTTP headers.
+// Other maps like ARGS are case sensitive, "User != user"
+// But a query will always return the original case sensitive key,
+// regarding of isCaseSensitiveKey
+func NewMap(variable variables.RuleVariable, isCaseSensitiveKey bool) *Map {
```

Comment:
Instead of a bool flag, for the public functions I think two factories `NewMap` and `NewMapCaseSensitiveKeys` or similar would be easier to understand.